gpbft/chain.go

```diff
@@ -425,6 +425,51 @@ func (c *ECChain) Key() ECChainKey {
 	return c.key
 }
 
+// KeysForPrefixes return batch of keys for all prefixes
+// the indexes to this array correspond to Prefix(i) calls
+func (c *ECChain) KeysForPrefixes() []ECChainKey {
+	if c.IsZero() {
+		return nil
+	}
+	values := make([][]byte, c.Len())
+	for i, ts := range c.TipSets {
+		values[i] = ts.MarshalForSigning()
+	}
+
+	batch := merkle.BatchTree(values)
+	res := make([]ECChainKey, c.Len())
+	for i := 0; i < c.Len(); i++ {
```

Comment:
```suggestion
	for i := range c.Len() {
```

Reply:
How did I miss for years that you can do `for i := range <int>`??

Reply:
It's new (since 1.22).